grails-app/domain/org/broadinstitute/orsp/Issue.groovy

```diff
@@ -205,4 +207,14 @@ class Issue {
         extraproperties
     }
 
+    /** AttachmentApproved verifies if there's any attachment status different to 'Approved'
+     *  if there is any, it will return a value different to null, meaning it still has attachments unReviewed
+     *
+     * @return Boolean indicating if all attachments have 'Approved' status
+     */
+    transient Boolean attachmentApproved(){
+        getAttachments()?.find {
+            it.status != IssueStatus.Approved.getName()
+        } == null
```

Comment:
Nit: Very minor suggestion. You could also use collect and query the emptiness of the collection. I usually like that approach over checking against null. Definitely not required.
```
        getAttachments()?.collect {
            it.status != IssueStatus.Approved.getName()
        }?.isEmpty()
```
Alternatively, you could use the spread operator over status in combination with `any` which is nice shorthand:
```
getAttachments()?.status?.any { 
    !it.equals(IssueStatus.Approved.getName()) 
}
```
This is an area where Groovy shines and comes closest to a scala-like functional style.

Reply:
Good to know! I used the second suggestion with the spread operator.